evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -524,6 +548,145 @@ impl<F: RichField> State<F> for Interpreter<F> {
         }
     }
 
+    fn check_against_struct_logs_before_op(
+        &mut self,
+        opcode: u8,
+        is_user_mode: bool,
+    ) -> Result<(), ProgramError> {
+        if let Some(struct_logs) = &self.struct_logs
+            && is_user_mode
+            && log_enabled!(log::Level::Debug)
+        {
+            let txn_idx = self.generation_state.next_txn_index;
+
+            if let Some(txn_struct_logs) = &struct_logs[txn_idx - 1] {
+                let counter = self.struct_log_debugger_info.counter;
+                if counter == 0 {
+                    // Initialize txn gas.
+                    let gas_limit_address = MemoryAddress::new(
+                        self.get_registers().context,
+                        Segment::ContextMetadata,
+                        ContextMetadata::GasLimit.unscale(), // context offsets are already scaled
+                    );
+                    let gas_limit = self.generation_state.get_from_memory(gas_limit_address);
+                    self.struct_log_debugger_info.gas = gas_limit.as_usize();
+                    // Check against actual initial gas.
+                    if gas_limit.as_u64() != txn_struct_logs[0].gas {
+                        log::warn!("Wrong initial transaction gas: it is {:?} in the Kernel but {:?} in the struct logs.", gas_limit.as_u64(), txn_struct_logs[0].gas);
```

Comment:
The log is slightly too verbose, maybe we could do something like
```suggestion
                        log::warn!("Wrong Initial txn gas: expected {:?}, got {:?}.", txn_struct_logs[0].gas,  gas_limit.as_u64());
```